agent/agent.go

```diff
@@ -273,3 +290,16 @@ func generateAgentID() string {
 	}
 	return agentId.String()
 }
+
+func parseDSN(dsnString string) (apiKey string, apiEndpoint string, err error) {
```

Comment:
A unit test for this function would be nice

Reply:
ok

Reply:
done